From: Andrew Morton <akpm@linux-foundation.org>

Cc: Paul Menage <menage@google.com>
Cc: William Lee Irwin III <wli@holomorphy.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 init/main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff -puN init/main.c~containersv10-basic-container-framework-fix-2 init/main.c
--- a/init/main.c~containersv10-basic-container-framework-fix-2
+++ a/init/main.c
@@ -512,7 +512,6 @@ asmlinkage void __init start_kernel(void
 	char * command_line;
 	extern struct kernel_param __start___param[], __stop___param[];
 
-	container_init_early();
 	smp_setup_processor_id();
 
 	/*
@@ -521,6 +520,7 @@ asmlinkage void __init start_kernel(void
 	 */
 	unwind_init();
 	lockdep_init();
+	container_init_early();
 
 	local_irq_disable();
 	early_boot_irqs_off();
_
