From: Daniel Walker <dwalker@mvista.com>

Signed-off-by: Daniel Walker <dwalker@mvista.com>
Cc: Paul Menage <menage@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/cpuset.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN kernel/cpuset.c~containersv10-make-cpusets-a-client-of-containers-whitespace kernel/cpuset.c
--- a/kernel/cpuset.c~containersv10-make-cpusets-a-client-of-containers-whitespace
+++ a/kernel/cpuset.c
@@ -513,7 +513,7 @@ static int update_cpumask(struct cpuset 
 	cpus_and(trialcs.cpus_allowed, trialcs.cpus_allowed, cpu_online_map);
 	/* cpus_allowed cannot be empty for a cpuset with attached tasks. */
 	if (container_task_count(cs->css.container) &&
-            cpus_empty(trialcs.cpus_allowed))
+	    cpus_empty(trialcs.cpus_allowed))
 		return -ENOSPC;
 	retval = validate_change(cs, &trialcs);
 	if (retval < 0)
_
