From: Andrew Morton <akpm@linux-foundation.org>

Cc: "Eric W. Biederman" <ebiederm@xmission.com>
Cc: "Serge E. Hallyn" <serue@us.ibm.com>
Cc: Balbir Singh <balbir@in.ibm.com>
Cc: Dave Hansen <haveblue@us.ibm.com>
Cc: Herbert Poetzl <herbert@13thfloor.at>
Cc: Kirill Korotaev <dev@openvz.org>
Cc: Paul Jackson <pj@sgi.com>
Cc: Paul Menage <menage@google.com>
Cc: Srivatsa Vaddagiri <vatsa@in.ibm.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/container.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff -puN kernel/container.c~containersv10-add-container_clone-interface-fix kernel/container.c
--- a/kernel/container.c~containersv10-add-container_clone-interface-fix
+++ a/kernel/container.c
@@ -1716,18 +1716,18 @@ int container_clone(struct task_struct *
 }
 
 /* See if "cont" is a descendant of the current task's container in
- * the appropriate hierarchy */
-
+ * the appropriate hierarchy
+ */
 int container_is_descendant(const struct container *cont)
 {
 	int ret;
 	struct container *target;
 	int subsys_id;
+
 	get_first_subsys(cont, NULL, &subsys_id);
 	target = task_container(current, subsys_id);
-	while (cont != target && cont!= cont->top_container) {
+	while (cont != target && cont!= cont->top_container)
 		cont = cont->parent;
-	}
 	ret = (cont == target);
 	return ret;
 }
_
